homeassistant/components/sonos/media_player.py

```diff
@@ -751,17 +751,23 @@ async def async_browse_media(
             media_content_type,
         )
 
-    def join_players(self, group_members):
+    async def async_join_players(self, group_members):
         """Join `group_members` as a player group with the current player."""
-        speakers = []
-        for entity_id in group_members:
-            if speaker := self.hass.data[DATA_SONOS].entity_id_mappings.get(entity_id):
-                speakers.append(speaker)
-            else:
-                raise HomeAssistantError(f"Not a known Sonos entity_id: {entity_id}")
-
-        self.speaker.join(speakers)
-
-    def unjoin_player(self):
+        async with self.hass.data[DATA_SONOS].topology_condition:
```

Comment:
you write `lock` in description but use a condition here

Reply:
I was trying to keep the description simple as the way it's used here _should_ effectively act like a simple lock. Am I missing an angle on why it shouldn't be used in this manner?

Reply:
was just checking!